control_plane/attachment_service/src/service.rs

```diff
@@ -2260,7 +2260,15 @@ impl Service {
 
         // unwrap safety: we would have returned above if we didn't find at least one shard to split
         let old_shard_count = old_shard_count.unwrap();
-        let shard_ident = shard_ident.unwrap();
+        let shard_ident = if let Some(new_stripe_size) = split_req.new_stripe_size {
+            // This ShardIdentity will be used as the template for all children, so this implicitly
+            // applies the new stripe size to the children.
+            let mut shard_ident = shard_ident.unwrap();
+            shard_ident.stripe_size = new_stripe_size;
```

Comment:
Would be nice to assert `shard_ident.stripe_size == 1` here.

Reply:
Yes, this really should be a validation site -- we would otherwise proceed to hit the validation code on the pageserver side.  Added an ApiError::BadRequest path here (no need to assert)